include/ada/url.h

```diff
@@ -120,7 +115,7 @@ namespace ada {
     bool set_host_or_hostname(std::string_view input, bool override_hostname);
 
     /** @private */
-    void update_base_hash(const std::string_view input);
+    void update_base_hash(std::optional<std::string> input);
```

Comment:
You had update_base_hash(std::string view input) which looked good and fast. The new function does not look nearly as lightweight. This is unfortunate.

Reply:
hash/fragment is optional and the `href` representation as well as `getter` representation changes according to having a value or not, since I don't see any other usage than using std::optional. The previous implementation was wrong.